proposals/stdlib/instant.md

```diff
@@ -0,0 +1,1365 @@
+println("You went without food for ${lunchStarted - breakfastEnded}")
+```
+
+#### `toString` and `parse`
+
+`toString` and `parse` use the format described in ISO-8601-1:2019, 5.4.2.1b),
+excluding the format without the offset.
+The format used is also mostly compatible with
+<https://www.rfc-editor.org/rfc/rfc3339#section-5.6>.
+Both of these formats are widely used, but also human-readable.
+
+Example: `2024-08024T16:22:34Z`.
+
+##### Advanced parsing and formatting
+
+There are other important formats for instant values, and also, there are
+limitations to the default format and how it can be used:
+
+* People may want to format RFC 1123 strings, like
+  `Mon, 30 Jun 2008 11:05:30 -0300`.
+* Occasionally, `24:00:00` is used to denote the last moment of the day.
+* `23:59:60` can represent positive leap seconds.
+* Given an ISO string like `2024-08-20T16:26:15+02:00`, someone may want to
+  obtain the `+02:00` part and not just the `Instant` value.
+* Someone may want to format an `Instant` with a UTC offset other than 0.
+
+All of this is outside the scope of `kotlin.time.Instant`: inherently,
+`kotlin.time.Instant` doesn't "know" anything about days or months or what hour
+it is. These concepts are strictly in the (calendar- and timezone-aware)
+datetime and not physical time territory.
+Therefore, advanced parsing and formatting needs are fulfilled in
+`kotlinx-datetime`.
+If the Standard Library ever acquires a flexible enough API for defining custom
+formats, we may want to revisit this.
+
+##### Leap seconds
+
+A departure from both RFC 3339 and ISO 8601 is that we do not allow parsing
+positive leap seconds (like `23:59:60Z`) and do not forbid parsing
+`23:59:59Z` even when there is a negative leap second.
+The reason for this is our choice of the UTC-SLS time scale: in that time scale,
+`Instant` values with `23:59:60Z` do not exist, and `23:59:59Z` never gets
+omitted, and instead, the extra and missing seconds are emulated by speeding
+the clock up and down.
+
+Java supports parsing `23:59:60Z`, but when the offset is different from `Z`,
+it does so incorrectly as of writing, allowing inserting leap seconds at
+`23:59:60` regardless of the UTC offset and forbidding the second value `:60`
+in all other combinations.
+
+This may be an issue for compatibility: if another system produces
+`23:59:60Z`, we will not be able to parse that.
+In that case, the user can employ the parsing and formatting API in
+`kotlinx-datetime` to decide how they want to handle leap seconds.
+It's also worth noting that we didn't find a significant demand for leap
+second handling in other parsing and formatting APIs, so this concern is mostly
+theoretical.
+
+#### The deprecated `now`
+
+A common way to obtain the current `Instant` on the JVM is to call
+`Instant.now()`:
+
+* <https://grep.app/search?q=Instant.now%28%29> is 13000 hits as of writing.
+* <https://grep.app/search?q=Clock.systemUTC%28%29.instant%28%29> is less than
+  100 hits.
```

Comment:
Not sure this comparison is useful here since these expressions are equivalent. The point of the `Clock` abstraction is instead to facilitate inversion of control as noted in its section below. You'd have to compare it to those calling `Clock::instant` which is much harder to search for. If we assume people name their injected `Clock` as "clock", you get 1200 hits: https://grep.app/search?q=clock.instant%28%29. That's not exhaustive, but it's probably a majority.